tests/core/persistent-journal-namespace/task.yaml

```diff
@@ -0,0 +1,61 @@
+summary: Test journal.persistent core config option is honored by journal namespaces
+
+# these systems do not support journal quota groups due to their old systemd versions.
+# requires systemd v245+
+systems:
+  - -ubuntu-core-16-*
+  - -ubuntu-core-18-*
+
+prepare: |
+  systemctl status snapd > before.txt
+  "$TESTSTOOLS"/snaps-state install-local test-snapd-journal-quota
+  snap set system experimental.quota-groups=true
+  tests.cleanup defer snap unset system experimental.quota-groups
+
+restore: |
+  # disable persistent journal
+  rm -rf /var/log/journal
+  systemctl kill --signal=SIGUSR1 systemd-journald
+    
+execute: |
+  echo "Wait for first boot to be done"
+  snap wait system seed.loaded
+
+  echo "Precondition check, persistent journal is not available by default"
+  not test -e /var/log/journal
+
+  echo "Check that persistent journal can be enabled"
+  snap set core journal.persistent=true
+  test -e /var/log/journal/
+  test -e /var/log/journal/.snapd-created
+
+  # Create a journal quota group and set arbitrary limits
+  echo "Create a journal namespace with the logger snap in it"
+  snap set-quota group-one --journal-size=16MB --journal-rate-limit=100/10ms test-snapd-journal-quota
+
+  # this check relies on *anything* getting logged; enabling persistent
+  # journal writes an entry about journal size, which should be sufficient.
+  MACHINE_ID=$(cat /etc/machine-id)
+  retry -n30 --wait 1 test -e "/var/log/journal/$MACHINE_ID"
+
+  echo "Starting service to fill namespace with logs"
+  snap start test-snapd-journal-quota.logger
+
+  # Verify the presence of the namespace log, and do this with retries as
+  # the logger writes onces every second
+  retry -n 5 sh -c "test -e /var/log/journal/$MACHINE_ID.snap-group-one"
+
+  echo "Check that persistent journal can be disabled"
+  snap set core journal.persistent=false
+  not test -e /var/log/journal
+
+  # Verify that, over a few seconds, nothing is created
+  # to make sure the namespace is not recreating the path
+  if retry -n 3 sh -c "test -e /var/log/journal"; then
+    echo "Journal path was recreated, it shouldn't have happened"
+    exit 1
+  fi
```

Comment:
Can we try only once, after a shorter time (maybe half a second)? This is making the spread tests take 3 seconds longer.

Reply:
Half a second is not enough, the service only writes every second. I can reduce it to 2 retries.